src/host/libpixyusb_swig/pixy_viewer.py

```diff
@@ -557,35 +571,57 @@ def __init__(self):
         self.config(menu=self._window.get_menubar())
         self.mainloop()
 
+    def is_exit(self):
+        return self._window.is_exit()
+
+
+## Wait for pixy to be connected
+def wait_pixy():
+    if pixy.pixy_init() == 0:
+        return
+
+    print("\n-- No Pixy detected. Plug a Pixy to the USB port. --")
+    while True:
+        if pixy.pixy_init() == 0:
+            break
+        time.sleep(1)
+
 
 ## Main function to capture images from pixy camera
 def main():
     print("MTTR Pixy Viewer " + VERSION)
 
-    # Initialize Pixy interface
-    if pixy.pixy_init() != 0:
-        print("Failed to initialize USB interface")
-        sys.exit(-1)
+    while True:
+        # Wait for Pixy to be connected
+        wait_pixy()
+
+        # Stop default program
+        if pixy.pixy_stop() != 0:
+            print("Failed to stop Pixy")
+            pixy.pixy_close()
+            continue
+
+        # Print pixy's firmware verssion
+        _, major, minor, build = pixy.pixy_get_firmware_version()
+        print("Firmware version %d.%d.%d" % (major, minor, build))
```

Comment:
Instead of printing, you can  consider using python logging module in the future as these messages are more sustainable, can be written to log files, etc. https://docs.python.org/3/library/logging.html. Out of scope of this PR but may be useful

Reply:
The output prints at 20ms intervals. I'll have to make sure the overhead doesn't jeopardize the performance.